### DISCUSSION TITLE
Unable to DeDRM ADE epub

### DISCUSSION THREAD

**Discussion by Ossiiris** [apprenticeharper/DeDRM_tools#1752] [Q&A]
Dear, on a complete fresh Windows 10 install, I am unable to DeDRM ADE epubs. Have ADE 2.0.1 installed on the same machine as Calibre. ADE 2.0.1. can open the book(s) without issues. Have Calibre 5.22.1 and DeDRM 7.2.1. What I noticed is that there are no ADE keys when I open the plugin.

Atteched the debug file from Calibre.

[tmp5d7v3er1.txt](https://github.com/apprenticeharper/DeDRM_tools/files/6722161/tmp5d7v3er1.txt)

Have read through a lot of posts but so far nothing helped. Any ideas what I should do to make it work?

Thank you.

**Comment by ElleKayEm** [comment]
You have ADE authorized with your Adobe ID?

**Comment by Ossiiris** [comment]
Yes, I have. Without it, ADE would also not open the book.

Op ma 28 jun. 2021 04:07 schreef ElleKayEm ***@***.***>:

**Reply by ElleKayEm** [reply]
An acsm would open, but the epub would be tied to that installation of ADE instead of an Adobe ID.  And it wouldn't work will DeDRM.

I see this error in the log:
>line 349, in adeptkeys
calibre_plugins.dedrm.adobekey.ADEPTError: PyCrypto or OpenSSL must be installed

Which is weird because I don't know why that would be necessary with Windows calibre.  I've only seen that with Linux calibre.

Not sure what to suggest.  Maybe a reinstall of calibre or 64-bit calibre.

**Comment by Ossiiris** [comment]
Hi,

thank you for your replies. Unfortunately I already (re-)installed the 64-bit version several times. Besides, I have exactly the same issue on a similar but different W10 pc (not a fresh W10 install).

Yesterday, I also noticed that line and tried to install PyCrypto, but either I am doing something wrong, either it does not help. Downloaded and installed Python 2.7 from Download & Install Python | ActiveState [https://www.activestate.com/products/python/downloads/] + PyCrypto for 2.7.

From what I have read DeDRm should generate a default ADE key, but in the configuration of the plugin , there is nothing there.

Any other suggestions?
Op 28-6-2021 13:40:16, ElleKayEm ***@***.***> schreef:
An acsm would open, but the epub would be tied to that installation of ADE instead of an Adobe ID. And it wouldn't work will DeDRM.
I see this error in the log:
line 349, in adeptkeys
calibre_plugins.dedrm.adobekey.ADEPTError: PyCrypto or OpenSSL must be installed
Which is weird because I don't know why that would be necessary with Windows calibre. I've only seen that with Linux calibre.
Not sure what to suggest. Maybe a reinstall of calibre or 64-bit calibre.

**Reply by ElleKayEm** [reply]
Ok your log showed 32-bit calibre.  It shouldn't be necessary to install Python or PyCrypto with Windows calibre.  It should use calibre's embedded Python.  And these versions of calibre and DeDRM are Python 3.

DeDRM will save a key when it's successfully decrypted a book.  In your case, the script that retrieves the key doesn't seem to be working.

Try going back to calibre 4.23 and DeDRM 6.8.1 (the last Python 2 versions).  After the key is saved, you might be able to update back to the current versions.  I know I'm having no issues with ADE epubs on Windows 10 and the current versions.  But I had previously used those older versions - not a fresh install.

**Reply by DRWolff1954** [reply]
I (and others) am having a similar problem, as described in:
DeDRM Decryption of EPUB in Calibre failed - Exception when getting Adobe Key #1798.
Any help would be most appreciated!
Thanks

**Comment by Ridderxxl** [comment]
Hi there,

Thank you for the thread here, it describes my problem nearly exactly. 
My log varies a bit from the ones here in that it seems like Calibre does not even try to decrypt the books. (log below)
As you can see in the log i have the latest version of DeDrm, and Calibre. I am also running version ADE_4.5. During the time Calibre was open, in debug mode, I opened both books from Cablibre (double clicking) and they opened up in ADE without a hitch (opening ADE, connecting to the google books servers and downloading the whole book).

Am I missing something or describing the same problem as you are having?

Thanks in advance!

calibre Debug log
calibre 5.26 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.17763 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.17763')
Python 3.8.5
Windows: ('10', '10.0.17763', '', 'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (7, 2, 1) && Obok DeDRM (7, 2, 1)
calibre 5.26 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.17763 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.17763')
Python 3.8.5
Windows: ('10', '10.0.17763', '', 'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: DeDRM (7, 2, 1) && Obok DeDRM (7, 2, 1)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 68.07370184254607 x 56.09166666666667
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.50] splash screen shown
[0.50] Initializing db...
[0.50] db initialized
[0.50] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
[1.27] main UI initialized...
[1.27] Hiding splash screen
Starting QuickView
[29.58] splash screen hidden
[29.58] Started up in 29.58 seconds with 2 books
Added A Gentleman in Moscow-epub to db in: 0.4
Added 1 books in 0.8 seconds
Added La forja de un rebelde-epub to db in: 0.4
Added 1 books in 0.8 seconds

**Reply by Ridderxxl** [reply]
Oke! So the trick for me was:

1. install Calibre
2. install DeDrm plugin
3. close Calibre
4. open Calibre
5. open Digital editions
6. Left click one of the book thumbnails and select "show in folder"
7. Drag the books from THIS folder into Calibre.
Works like a charm! Thank you very much.


before I dragged the .ascm files that I downloaded from Google books straight from my downloads folder into Calibre, which does not work.

**Comment by DRWolff1954** [comment]
G'day
Very tardy reply, and it seems you have already found a solution. As did I,
but yours is much "cleaner".
The workaround I found was to download a book from Kobo into Kobo Desktop;
and then add it to Calibre using the OBEK DeDRM add-in.

Am I correct that somewhere in in your solution, *before* step 5, is:

   - install ADE (which version did you use?)
   - acquire ascm file (what source worked for your?)
   - import/download ebook into ADE

Thanks
Denis

**Reply by ElleKayEm** [reply]
It is recommended to use ADE 2.0.1 because later versions can handle DRM which cannot be removed (yet).  If the ebook store is not using this "hardened" DRM, you'll be fine.  But you won't know.  If it can be downloaded and read via ADE 2.0.1, then it has DRM which can be removed.  Any ebook store which makes their books available with Adobe DRM is a source for acsm files (Kobo, Google).  Before opening the acsm file with ADE, be sure to authorize ADE with your Adobe ID.
